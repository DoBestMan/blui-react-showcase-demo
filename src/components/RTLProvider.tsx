import React, { useEffect } from 'react';
import { ThemeProvider, createTheme, StyledEngineProvider } from '@mui/material/styles';
import { create } from 'jss';
import rtl from 'jss-rtl';
// import ThemeProvider from '@mui/styles/ThemeProvider';
import jssPreset from '@mui/styles/jssPreset';
import StylesProvider from '@mui/styles/StylesProvider';
import * as BLUIThemes from '@brightlayer-ui/react-themes';
import { useSelector } from 'react-redux';
import { AppStore } from '../__types__';

// Configure JSS
const jss = create({ plugins: [...jssPreset().plugins, rtl()] });

export const RTLThemeProvider: React.FC = (props) => {
    const dir = useSelector((store: AppStore) => store.app.direction);
    const theme = useSelector((store: AppStore) => store.app.theme);

    useEffect(() => {
        document.body.dir = dir;
    }, [dir]);

    return (
        <StylesProvider jss={jss}>
            <StyledEngineProvider injectFirst>
                <ThemeProvider
                    theme={createTheme(
                            Object.assign(theme === 'light' ? BLUIThemes.blue : BLUIThemes.blueDark, {
                                direction: dir,
                            })
                    )}
                >
                    {props.children}
                </ThemeProvider>
            </StyledEngineProvider>
        </StylesProvider>
    );
};
