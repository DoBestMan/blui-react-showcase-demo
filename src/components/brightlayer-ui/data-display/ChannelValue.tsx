import React from 'react';
import Typography from '@mui/material/Typography';
import { styled } from '@mui/material/styles';
import { ChannelValue } from '@brightlayer-ui/react-components';
import TrendingUp from '@mui/icons-material/TrendingUp';
import * as colors from '@brightlayer-ui/colors';
import { useSelector } from 'react-redux';
import { AppStore } from '../../../__types__';
import Box from '@mui/material/Box';

const ComponentContainer = styled(Box)(({ theme }) => ({
    display: 'flex',
    flexDirection: 'column',
    flex: 1,
    marginBottom: theme.spacing(4),
}));

const SectionTitle = styled(Typography)(({ theme }) => ({
    marginBottom: theme.spacing(2),
}));

export const ChannelValueExample: React.FC = () => {
    const direction = useSelector((store: AppStore) => store.app.direction);
    const rtl = direction === 'rtl';

    return (
        <>
            <ComponentContainer>
                <SectionTitle variant={'body1'}>Basic Usage</SectionTitle>
                <ChannelValue value={'123'} />
            </ComponentContainer>
            <ComponentContainer>
                <SectionTitle variant={'body1'}>w/ Units</SectionTitle>
                <ChannelValue value={'123'} units={'hz'} />
            </ComponentContainer>
            <ComponentContainer>
                <SectionTitle variant={'body1'}>w/ Icon</SectionTitle>
                <ChannelValue
                    value={'123'}
                    units={'hz'}
                    icon={<TrendingUp htmlColor={colors.red[500]} sx={rtl ? { transform: 'scaleX(-1)' } : {}} />}
                />
            </ComponentContainer>
            <ComponentContainer>
                <SectionTitle variant={'body1'}>w/ Custom Font Size</SectionTitle>
                <ChannelValue
                    value={'123'}
                    units={'hz'}
                    fontSize={24}
                    icon={<TrendingUp htmlColor={colors.red[500]} sx={rtl ? { transform: 'scaleX(-1)' } : {}} />}
                />
            </ComponentContainer>
        </>
    );
};
