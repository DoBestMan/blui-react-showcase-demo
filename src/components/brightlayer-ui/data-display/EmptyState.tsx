import React from 'react';
import Button from '@mui/material/Button';
import Typography from '@mui/material/Typography';
import Add from '@mui/icons-material/Add';
import Devices from '@mui/icons-material/Devices';
import LocationOff from '@mui/icons-material/LocationOff';
import NotListedLocation from '@mui/icons-material/NotListedLocation';
import { styled } from '@mui/material/styles';

import { EmptyState } from '@brightlayer-ui/react-components';
import { useSelector } from 'react-redux';
import { AppStore } from '../../../__types__';
import Box from '@mui/material/Box';

const Container = styled(Box)(({ theme }) => ({
    marginBottom: theme.spacing(4),
}));

const SectionTitle = styled(Typography)(({ theme }) => ({
    marginBottom: theme.spacing(2),
}));

export const EmptyStateExample: React.FC = () => {
    const direction = useSelector((store: AppStore) => store.app.direction);
    const rtl = direction === 'rtl';

    return (
        <>
            <Container>
                <SectionTitle variant={'body1'}>Basic Usage</SectionTitle>
                <EmptyState
                    icon={<NotListedLocation fontSize={'inherit'} sx={rtl ? { transform: 'scaleX(-1)' } : {}} />}
                    title={'Location Unknown'}
                />
            </Container>
            <Container>
                <SectionTitle variant={'body1'}>w/ Description</SectionTitle>
                <EmptyState
                    icon={<LocationOff fontSize={'inherit'} sx={rtl ? { transform: 'scaleX(-1)' } : {}} />}
                    title={'Location Services Disabled'}
                    description={'Enable Location Services via Settings to receive GPS information'}
                />
            </Container>
            <SectionTitle variant={'body1'}>w/ Actions</SectionTitle>
            <EmptyState
                icon={<Devices fontSize={'inherit'} sx={rtl ? { transform: 'scaleX(-1)' } : {}} />}
                title={'No Devices'}
                description={'Check your network connection or add a new device'}
                actions={
                    <Button variant={'outlined'} color={'primary'} startIcon={<Add />}>
                        Add Device
                    </Button>
                }
            />
        </>
    );
};
