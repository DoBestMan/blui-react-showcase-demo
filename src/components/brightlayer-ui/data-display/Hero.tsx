import React from 'react';
import Typography from '@mui/material/Typography';
import { ChannelValue, Hero, HeroBanner } from '@brightlayer-ui/react-components';
import GradeA from '@brightlayer-ui/icons-mui/GradeA';
import GradeB from '@brightlayer-ui/icons-mui/GradeB';
import Temp from '@brightlayer-ui/icons-mui/Temp';
import Schedule from '@mui/icons-material/Schedule';
import * as colors from '@brightlayer-ui/colors';
import { styled } from '@mui/material/styles';
import Box from '@mui/material/Box';

const ExampleRow = styled(Box)(({ theme }) => ({
    display: 'flex',
    justifyContent: 'space-evenly',
    marginBottom: theme.spacing(2),
}));

const ComponentContainer = styled(Box)(() => ({
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    flex: 1,
}));

const Label = styled(Typography)(({ theme }) => ({
    marginBottom: theme.spacing(1),
}));

export const HeroExample: React.FC = () => (
    <>
        <ExampleRow>
            <ComponentContainer>
                <Label variant={'body2'}>Basic Usage</Label>
                <Hero
                    icon={<GradeA fontSize={'inherit'} />}
                    label={'Efficiency'}
                    sx={{ flex: 1 }}
                    style={{ flex: 1 }}
                />
            </ComponentContainer>
            <ComponentContainer>
                <Label variant={'body2'}>w/ Value and Units</Label>
                <Hero
                    icon={<GradeB fontSize={'inherit'} />}
                    label={'Efficiency'}
                    ChannelValueProps={{ value: 88, units: '%' }}
                    style={{ flex: 1 }}
                />
            </ComponentContainer>
        </ExampleRow>
        <ExampleRow>
            <ComponentContainer>
                <Label variant={'body2'}>w/ Channel Value Children</Label>
                <Hero icon={<Schedule fontSize={'inherit'} />} label={'Efficiency'} style={{ flex: 1 }}>
                    <ChannelValue fontSize={'large'} value={1} units={'h'} />
                    <ChannelValue fontSize={'large'} value={27} units={'m'} />
                </Hero>
            </ComponentContainer>
            <ComponentContainer>
                <Label variant={'body2'}>w/ Icon Colors</Label>
                <Hero
                    iconBackgroundColor={colors.red[500]}
                    icon={<Temp fontSize={'inherit'} htmlColor={colors.white[50]} />}
                    label={'Temperature'}
                    ChannelValueProps={{ value: 38, units: '°C' }}
                    style={{ flex: 1 }}
                />
            </ComponentContainer>
        </ExampleRow>
        <ComponentContainer>
            <Label variant={'body2'}>Within a HeroBanner</Label>
            <HeroBanner style={{ width: 376, border: `solid 1px ${colors.gray[50]}`, borderRadius: 4 }}>
                <Hero
                    icon={<GradeA fontSize={'inherit'} />}
                    label={'Efficiency'}
                    ChannelValueProps={{ value: '95', units: '%' }}
                />
                <Hero
                    icon={<GradeB fontSize={'inherit'} />}
                    label={'Efficiency'}
                    ChannelValueProps={{ value: '88', units: '%' }}
                />
                <Hero icon={<Schedule fontSize={'inherit'} />} label={'Efficiency'}>
                    <ChannelValue fontSize={'large'} value={1} units={'h'} />
                    <ChannelValue fontSize={'large'} value={27} units={'m'} />
                </Hero>
                <Hero
                    iconBackgroundColor={colors.red[500]}
                    icon={<Temp fontSize={'inherit'} htmlColor={colors.white[50]} />}
                    label={'Temperature'}
                    ChannelValueProps={{ value: 38, units: '°C' }}
                />
            </HeroBanner>
        </ComponentContainer>
    </>
);
