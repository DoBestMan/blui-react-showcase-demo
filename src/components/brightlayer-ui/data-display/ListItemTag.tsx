import React from 'react';
import Typography from '@mui/material/Typography';
import { useTheme, styled } from '@mui/material/styles';
import { InfoListItem, ListItemTag } from '@brightlayer-ui/react-components';
import * as colors from '@brightlayer-ui/colors';
import BrightnessMedium from '@mui/icons-material/BrightnessMedium';
import { useSelector } from 'react-redux';
import { AppStore } from '../../../__types__';
import Box from '@mui/material/Box';

const ExampleRow = styled(Box)(({ theme }) => ({
    display: 'flex',
    justifyContent: 'space-evenly',
    marginBottom: theme.spacing(2),
}));

const ComponentContainer = styled(Box)(() => ({
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    flex: 1,
}));

const Label = styled(Typography)(({ theme }) => ({
    marginBottom: theme.spacing(1),
}));

export const ListItemTagExample: React.FC = () => {
    const theme = useTheme();
    const direction = useSelector((store: AppStore) => store.app.direction);
    const rtl = direction === 'rtl';

    return (
        <>
            <ExampleRow>
                <ComponentContainer>
                    <Label variant={'body2'}>Basic Usage</Label>
                    <ListItemTag label={'active'} />
                </ComponentContainer>
                <ComponentContainer>
                    <Label variant={'body2'}>w/ Custom Colors</Label>
                    <ListItemTag label={'active'} backgroundColor={colors.red['500']} fontColor={colors.white['50']} />
                </ComponentContainer>
            </ExampleRow>
            <ComponentContainer style={{ marginTop: theme.spacing(6) }}>
                <Label variant={'body2'}>Within an Info List Item</Label>
                <InfoListItem
                    icon={<BrightnessMedium />}
                    title={'Info List Item'}
                    subtitle={'with List Item Tags'}
                    rightComponent={
                        <div style={{ display: 'flex' }}>
                            <ListItemTag
                                label={'Build Passing'}
                                backgroundColor={colors.green[300]}
                                fontColor={colors.black[900]}
                                style={{
                                    marginRight: rtl ? 0 : theme.spacing(2),
                                    marginLeft: rtl ? theme.spacing(2) : 0,
                                }}
                            />
                            <ListItemTag
                                label={'5 Bugs'}
                                backgroundColor={colors.red[300]}
                                fontColor={colors.black[900]}
                            />
                        </div>
                    }
                />
            </ComponentContainer>
        </>
    );
};
