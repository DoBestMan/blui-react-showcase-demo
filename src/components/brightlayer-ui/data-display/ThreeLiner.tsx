import React from 'react';
import useTheme from '@material-ui/core/styles/useTheme';
import createStyles from '@material-ui/core/styles/createStyles';
import makeStyles from '@material-ui/core/styles/makeStyles';
import { ChannelValue, ThreeLiner } from '@brightlayer-ui/react-components';
import TrendingUp from '@material-ui/icons/TrendingUp';
import * as colors from '@brightlayer-ui/colors';
import { useSelector } from 'react-redux';
import { AppStore } from '../../../__types__';
import clsx from 'clsx';

const useStyles = makeStyles(() =>
    createStyles({
        iconFlip: {
            transform: 'scaleX(-1)',
        },
    })
);

export const ThreeLinerExample: React.FC = () => {
    const theme = useTheme();
    const classes = useStyles();
    const direction = useSelector((store: AppStore) => store.app.direction);
    const rtl = direction === 'rtl';

    return (
        <>
            <ThreeLiner title={'Three Liner Component'} subtitle={'with basic usage'} info={'...and a third line'} />
            <ThreeLiner
                style={{ marginTop: theme.spacing(4) }}
                title={'Three Liner Component'}
                subtitle={'with custom content'}
                info={
                    <ChannelValue
                        value={'123'}
                        units={'hz'}
                        icon={<TrendingUp htmlColor={colors.red[500]} className={clsx({ [classes.iconFlip]: rtl })} />}
                    />
                }
            />
        </>
    );
};
