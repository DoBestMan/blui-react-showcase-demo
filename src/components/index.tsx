export * from './RTLProvider';
export * from './SharedAppBar';
export * from './material-ui';
export * from './ExampleCardList';
