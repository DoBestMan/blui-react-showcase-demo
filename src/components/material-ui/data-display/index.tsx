export * from './Avatar';
export * from './Badge';
export * from './Chip';
export * from './Divider';
export * from './Icon';
export * from './List';
export * from './Table';
export * from './Tooltip';
export * from './Typography';
