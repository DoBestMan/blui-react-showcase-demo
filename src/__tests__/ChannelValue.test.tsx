import React from 'react';
import { CssBaseline } from '@material-ui/core';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { store } from '../redux/store';
import { RTLThemeProvider } from '../components/RTLProvider';
import { ChannelValueExample } from '../components/px-blue/data-display';

describe('ChannelValue test ', () => {
    it('Component renders correctly', () => {
        const tree = renderer
            .create(
                <Provider store={store}>
                    <RTLThemeProvider>
                        <CssBaseline />
                        <ChannelValueExample />
                    </RTLThemeProvider>
                </Provider>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
