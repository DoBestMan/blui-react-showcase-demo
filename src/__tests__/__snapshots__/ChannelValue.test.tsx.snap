// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelValue test  ChannelValue component renders 1`] = `
Array [
  <div
    className="makeStyles-componentContainer-2"
  >
    <p
      className="MuiTypography-root makeStyles-sectionTitle-1 MuiTypography-body1"
    >
      Basic Usage
    </p>
    <span
      className="makeStyles-root-4 makeStyles-root-9"
      data-test="wrapper"
    >
      <h6
        className="MuiTypography-root makeStyles-text-6 makeStyles-value-8 MuiTypography-h6 MuiTypography-colorInherit"
        data-test="value"
      >
        123
      </h6>
    </span>
  </div>,
  <div
    className="makeStyles-componentContainer-2"
  >
    <p
      className="MuiTypography-root makeStyles-sectionTitle-1 MuiTypography-body1"
    >
      w/ Units
    </p>
    <span
      className="makeStyles-root-4 makeStyles-root-10"
      data-test="wrapper"
    >
      <h6
        className="MuiTypography-root makeStyles-text-6 makeStyles-value-8 MuiTypography-h6 MuiTypography-colorInherit"
        data-test="value"
      >
        123
      </h6>
      <h6
        className="MuiTypography-root makeStyles-text-6 makeStyles-units-7 MuiTypography-h6 MuiTypography-colorInherit"
        data-test="units"
      >
        hz
      </h6>
    </span>
  </div>,
  <div
    className="makeStyles-componentContainer-2"
  >
    <p
      className="MuiTypography-root makeStyles-sectionTitle-1 MuiTypography-body1"
    >
      w/ Icon
    </p>
    <span
      className="makeStyles-root-4 makeStyles-root-11"
      data-test="wrapper"
    >
      <span
        className="makeStyles-icon-5"
        data-test="icon"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          color="#ca3c3d"
          focusable="false"
          style={
            Object {
              "display": "block",
              "fontSize": "inherit",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M16 6l2.29 2.29-4.88 4.88-4-4L2 16.59 3.41 18l6-6 4 4 6.3-6.29L22 12V6z"
          />
        </svg>
      </span>
      <h6
        className="MuiTypography-root makeStyles-text-6 makeStyles-value-8 MuiTypography-h6 MuiTypography-colorInherit"
        data-test="value"
      >
        123
      </h6>
      <h6
        className="MuiTypography-root makeStyles-text-6 makeStyles-units-7 MuiTypography-h6 MuiTypography-colorInherit"
        data-test="units"
      >
        hz
      </h6>
    </span>
  </div>,
  <div
    className="makeStyles-componentContainer-2"
  >
    <p
      className="MuiTypography-root makeStyles-sectionTitle-1 MuiTypography-body1"
    >
      w/ Custom Font Size
    </p>
    <span
      className="makeStyles-root-4 makeStyles-root-12"
      data-test="wrapper"
    >
      <span
        className="makeStyles-icon-5"
        data-test="icon"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          color="#ca3c3d"
          focusable="false"
          style={
            Object {
              "display": "block",
              "fontSize": "inherit",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M16 6l2.29 2.29-4.88 4.88-4-4L2 16.59 3.41 18l6-6 4 4 6.3-6.29L22 12V6z"
          />
        </svg>
      </span>
      <h6
        className="MuiTypography-root makeStyles-text-6 makeStyles-value-8 MuiTypography-h6 MuiTypography-colorInherit"
        data-test="value"
      >
        123
      </h6>
      <h6
        className="MuiTypography-root makeStyles-text-6 makeStyles-units-7 MuiTypography-h6 MuiTypography-colorInherit"
        data-test="units"
      >
        hz
      </h6>
    </span>
  </div>,
]
`;

exports[`ChannelValue test  Component renders correctly 1`] = `
Array [
  <div
    className="makeStyles-componentContainer-2"
  >
    <p
      className="MuiTypography-root makeStyles-sectionTitle-1 MuiTypography-body1"
    >
      Basic Usage
    </p>
    <span
      className="makeStyles-root-4 makeStyles-root-9"
      data-test="wrapper"
    >
      <h6
        className="MuiTypography-root makeStyles-text-6 makeStyles-value-8 MuiTypography-h6 MuiTypography-colorInherit"
        data-test="value"
      >
        123
      </h6>
    </span>
  </div>,
  <div
    className="makeStyles-componentContainer-2"
  >
    <p
      className="MuiTypography-root makeStyles-sectionTitle-1 MuiTypography-body1"
    >
      w/ Units
    </p>
    <span
      className="makeStyles-root-4 makeStyles-root-10"
      data-test="wrapper"
    >
      <h6
        className="MuiTypography-root makeStyles-text-6 makeStyles-value-8 MuiTypography-h6 MuiTypography-colorInherit"
        data-test="value"
      >
        123
      </h6>
      <h6
        className="MuiTypography-root makeStyles-text-6 makeStyles-units-7 MuiTypography-h6 MuiTypography-colorInherit"
        data-test="units"
      >
        hz
      </h6>
    </span>
  </div>,
  <div
    className="makeStyles-componentContainer-2"
  >
    <p
      className="MuiTypography-root makeStyles-sectionTitle-1 MuiTypography-body1"
    >
      w/ Icon
    </p>
    <span
      className="makeStyles-root-4 makeStyles-root-11"
      data-test="wrapper"
    >
      <span
        className="makeStyles-icon-5"
        data-test="icon"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          color="#ca3c3d"
          focusable="false"
          style={
            Object {
              "display": "block",
              "fontSize": "inherit",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M16 6l2.29 2.29-4.88 4.88-4-4L2 16.59 3.41 18l6-6 4 4 6.3-6.29L22 12V6z"
          />
        </svg>
      </span>
      <h6
        className="MuiTypography-root makeStyles-text-6 makeStyles-value-8 MuiTypography-h6 MuiTypography-colorInherit"
        data-test="value"
      >
        123
      </h6>
      <h6
        className="MuiTypography-root makeStyles-text-6 makeStyles-units-7 MuiTypography-h6 MuiTypography-colorInherit"
        data-test="units"
      >
        hz
      </h6>
    </span>
  </div>,
  <div
    className="makeStyles-componentContainer-2"
  >
    <p
      className="MuiTypography-root makeStyles-sectionTitle-1 MuiTypography-body1"
    >
      w/ Custom Font Size
    </p>
    <span
      className="makeStyles-root-4 makeStyles-root-12"
      data-test="wrapper"
    >
      <span
        className="makeStyles-icon-5"
        data-test="icon"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          color="#ca3c3d"
          focusable="false"
          style={
            Object {
              "display": "block",
              "fontSize": "inherit",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M16 6l2.29 2.29-4.88 4.88-4-4L2 16.59 3.41 18l6-6 4 4 6.3-6.29L22 12V6z"
          />
        </svg>
      </span>
      <h6
        className="MuiTypography-root makeStyles-text-6 makeStyles-value-8 MuiTypography-h6 MuiTypography-colorInherit"
        data-test="value"
      >
        123
      </h6>
      <h6
        className="MuiTypography-root makeStyles-text-6 makeStyles-units-7 MuiTypography-h6 MuiTypography-colorInherit"
        data-test="units"
      >
        hz
      </h6>
    </span>
  </div>,
]
`;
